lib/src/hex_util.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+fn to_reverse_hex_digit(b: u8) -> Option<u8> {
+    let value = if (b'0'..=b'9').contains(&b) {
+        b - b'0'
+    } else if (b'A'..=b'F').contains(&b) {
+        b - b'A' + 10
+    } else if (b'a'..=b'f').contains(&b) {
+        b - b'a' + 10
+    } else {
+        return None;
+    };
+    Some(b'z' - value)
+}
+
+fn to_forward_hex_digit(b: u8) -> Option<u8> {
+    let value = if (b'k'..=b'z').contains(&b) {
+        b'z' - b
+    } else if (b'K'..=b'Z').contains(&b) {
+        b'Z' - b
+    } else {
+        return None;
+    };
+    if value < 10 {
+        Some(b'0' + value)
+    } else {
+        Some(b'a' + value - 10)
+    }
+}
+
+pub fn to_forward_hex(reverse_hex: &str) -> Option<String> {
+    let mut forward_hex = String::with_capacity(reverse_hex.len());
+    for b in reverse_hex.as_bytes() {
+        if let Some(forward_digit) = to_forward_hex_digit(*b) {
+            forward_hex.push(forward_digit as char);
+        } else {
+            return None;
+        }
+    }
+    Some(forward_hex)
+}
+
+pub fn to_reverse_hex(forward_hex: &str) -> Option<String> {
+    let mut reverse_hex = String::with_capacity(forward_hex.len());
+    for b in forward_hex.as_bytes() {
+        if let Some(forward_digit) = to_reverse_hex_digit(*b) {
+            reverse_hex.push(forward_digit as char);
+        } else {
+            return None;
+        }
+    }
+    Some(reverse_hex)
+}
```

Comment:
nit: `.bytes()` rather than `.as_bytes()` for iterating, and could be shorter in a more functional style

```rust
pub fn to_reverse_hex(forward_hex: &str) -> Option<String> {
    forward_hex
        .bytes()
        .map(|b| to_reverse_hex_digit(b).map(char::from))
        .collect::<Option<String>>()
}
```

or in an imperative style with `?`:

```rust
pub fn to_reverse_hex(forward_hex: &str) -> Option<String> {
    let mut reverse_hex = String::with_capacity(forward_hex.len());
    for b in forward_hex.bytes() {
        reverse_hex.push(to_reverse_hex_digit(b)? as char);
    }
    Some(reverse_hex)
}
```